src/core/main.c

```diff
@@ -1935,30 +1983,74 @@ static int do_queue_default_job(
         return 0;
 }
 
-static void free_arguments(void) {
-
-        /* Frees all arg_* variables, with the exception of arg_serialization */
-        rlimit_free_all(arg_default_rlimit);
+static void reset_arguments(void) {
+        /* Frees/resets arg_* variables, with a few exceptions commented below. */
 
         arg_default_unit = mfree(arg_default_unit);
+
+        /* arg_system — ignore */
+
+        arg_dump_core = true;
+        arg_crash_chvt = -1;
+        arg_crash_shell = false;
+        arg_crash_reboot = false;
         arg_confirm_spawn = mfree(arg_confirm_spawn);
         arg_join_controllers = strv_free_free(arg_join_controllers);
+        arg_show_status = _SHOW_STATUS_UNSET;
+        arg_switched_root = false;
+        arg_no_pager = false;
+        arg_service_watchdogs = true;
+        arg_default_std_output = EXEC_OUTPUT_JOURNAL;
+        arg_default_std_error = EXEC_OUTPUT_INHERIT;
+        arg_default_restart_usec = DEFAULT_RESTART_USEC;
+        arg_default_timeout_start_usec = DEFAULT_TIMEOUT_USEC;
+        arg_default_timeout_stop_usec = DEFAULT_TIMEOUT_USEC;
+        arg_default_timeout_abort_usec = DEFAULT_TIMEOUT_USEC;
+        arg_default_timeout_abort_set = false;
+        arg_default_start_limit_interval = DEFAULT_START_LIMIT_INTERVAL;
+        arg_default_start_limit_burst = DEFAULT_START_LIMIT_BURST;
+        arg_runtime_watchdog = 0;
+        arg_shutdown_watchdog = 10 * USEC_PER_MINUTE;
+        arg_early_core_pattern = NULL;
+        arg_watchdog_device = NULL;
+
         arg_default_environment = strv_free(arg_default_environment);
+        rlimit_free_all(arg_default_rlimit);
+
+        arg_capability_bounding_set = CAP_ALL;
+        arg_no_new_privs = false;
+        arg_timer_slack_nsec = NSEC_INFINITY;
+        arg_default_timer_accuracy_usec = 1 * USEC_PER_MINUTE;
+
         arg_syscall_archs = set_free(arg_syscall_archs);
+
+        /* arg_serialization — ignore */
+
+        arg_default_cpu_accounting = -1;
```

Comment:
This line right here is the culprit of the `test-seccomp` failure. Even though `arg_default_cpu_accounting` is zeroed-out, as it's in the BSS section, after the first systemd reload it's set to `-1` it, which equals to `true` after the boolean conversion, thus the CPU accounting is enabled system-wide.

The main issue here is that the CPU accounting is "forcefully" enabled even if you don't change the default in `/etc/systemd/system.conf`, because the respective line is commented out:

```
# grep DefaultCPUAccounting /etc/systemd/system.conf
#DefaultCPUAccounting=no
```
thus the unexpected behavior.

This can be easily reproduced, for example, in our `1minutetip`:

```
$ 1minutetip --flavor ci.c1.xlarge -n rhel8
...
## Following commands are executed inside the VM
# git clone https://github.com/systemd-rhel/rhel-8
# cd rhel-8/
# git fetch -fu origin refs/pull/41/merge:pr
# meson build
# ninja -C build install
# dracut -f --regenerate-all
# systemctl reboot
...
# systemctl show -p DefaultCPUAccounting
DefaultCPUAccounting=yes
# grep DefaultCPUAccounting /etc/systemd/system.conf
#DefaultCPUAccounting=no
```

Reply:
I suspect this may have been fixed in https://github.com/systemd/systemd/pull/10507/files#diff-108b33cf1bd0765d116dd401376ca356R755 where in case of `arg_default_cpu_accounting == -1` it would get overridden by a value returned from `cpu_accounting_is_cheap()`, which should return false when running on a system with cgroups v1.